facilitator/rust-toolchain.toml

```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.56.1"
+channel = "stable"
```

Comment:
I think we should revisit whether `rust-toolchain.toml` is a good idea. We should be running tests in GitHub CI with the same Rust toolchain version used to build containers. The version here is used for the former, the container image in the `prio-facilitator` Dockerfile sets the latter. The difficulty is that dependabot keeps our Dockerfile up to date, but doesn't appear to know about `rust-toolchain.toml`. We need to research what the current best practices are for specifying a minimum required Rust version that also plays nice with CI, building containers and dependabot.

Anyway, that's not really relevant to this PR, and using stable is strictly better than pinning to 1.56.1, I think.

Reply:
Agreed, though I'd bet a strategy of "pin everything to stable, until that causes a problem" would work out OK.

Reply:
A lot of projects specify a "minimum supported Rust version", or MSRV, colloquially. (This is largely a convention, but RFC 2495 will add a Rust version field to Cargo.toml) Some libraries go further, and make guarantees about for how many months/years their MSRV will trail the stable channel. There are differing opinions on whether bumping the MSRV implies different degrees of SemVer incompatibility. (As such, a dependency update may necessitate an MSRV bump)

In our situation, with the facilitator server being a binary, and libprio-rs presumably having few external consumers, we can probably afford to be more liberal with changing our MSRV. To hold ourselves to MSRV compatibility, we would want to at least run a check build in CI on both our specified MSRV and the latest stable channel. (and perhaps nightly) The main downside of attempting to support a wide range of Rust versions is that conflicts between lints or Clippy lints in the new version versus available APIs in the old version mean having to carry `#[allow()]` annotations in problem areas, or TODO comments where we would like to use a newer API, until the next MSRV bump.